docs/DeveloperGuide.md

```diff
@@ -1,15 +1,126 @@
+The UML diagram below illustrates the flow of how the application adds events:
+![Add Event Class Diagram](UML/Images/addEvent.png)
+
+
+The UML diagram below illustrates the flow of how the application adds modules:
+![Add Event Class Diagram](UML/Images/addModules.png)
+
+#### Why is the feature implemented this way?
+
+This component should be able to guide the user to inputting the correct format of the command to do data validation.
+It should be able to perform basic data validation checks to ensure that the user does not enter any invalid commands.
+This makes life easier to developers in the future if they wish to add new features that requires users to use new commands.
+
+### Storage Component
+API: `Storage.java`
+
+The storage component reads and writes user data from a local save in the form of a `.json` file.
+The Storage component:
+* Serializes and deserializes user data into a `.json` file format through the use of the Gson library
+* Loads data from `NusMods.json` into a HashMap for use by other classes.
+* Inherits from both `EventListStorage` and `NusModuleLoader`, and can be treated as either one.
+* Saves and loads information from the local hard disk
+* Depends on some classes (the `storage` component saves and retrieves objects)
+
+The class diagram below illustrates the structure of the storage package
+
+![Storage Class Diagram](UML/Images/StorageClass.png)
+
+#### How the feature is implemented:
+
+##### Load Events
+
+![Load Events Sequence Diagram](UML/Images/loadEvents.png)
+
+When the application starts up, the storage loadEvents() function will be called to load contents in the save file. 
+
+##### Load Modules
+![Load Modules Sequence Diagram](UML/Images/LoadModules.png)
+
+When any component requires reading the NUS module files, the loadModules() is called.
+
+##### Save Events
+
+![Save To File Sequence Diagram](UML/Images/SaveToFile.png)
+
+Similarly, the state of the user's event list is saved when the user exits the application by calling saveToFile().
+
+#### Justification for using gson
+The Gson library was chosen as it allowed for flexible adaptation of its TypeAdapter class, allowing for custom 
+serialization and deserialization of data to be saved. 
+
+### EventList component
+
+API: `EventList.java`
+
+this component maintains a list of Schedule instance. It receives commands from Parser.java and adds/deletes/edits tasks and their information in the list according to the commands.
+
+#### How is the feature implemented:
+
+the main functions are
+
+> - add new task (accepts event without starting time/ending time/ending date).
+> - delete a single tasks / delete all tasks.
+> - edit the time information of task (starting time/ending time/ending date can be omitted).
+> - search for a event by index / description.
+> - get all the detail of a event in the list in String form.
+
+The class diagram below illustrates the structure of the EventList component.
+
+<img src="UML\Images\EventListUML.png" />
```

Comment:
Perhaps the activation bar can be a single one as the object is created once and called again
![image](https://user-images.githubusercontent.com/88631870/228731838-4a02b932-ef15-4e75-a593-511109be0d81.png)